src/tree/param.h

```diff
@@ -73,6 +73,10 @@ struct TrainParam : public dmlc::Parameter<TrainParam> {
   bool refresh_leaf;
   // auxiliary data structure
   std::vector<int> monotone_constraints;
+  // structure to record interaction constraints
+  std::vector<int> int_constraints;
```

Comment:
To avoid confusion with "integer constraints", could we please name it interaction_constraints?